### DISCUSSION TITLE
Can I store container images on an external HDD while having Talos on the SD card (Raspberry Pi)

### DISCUSSION THREAD

**Discussion by kees-closed** [siderolabs/talos#9049] [General]
I'm trying to understand if this setup is possible: installing Talos on the SD card, but storing container images for Kubernetes on an external HDD. I'm using the Raspberry Pi 4. It seems possible to mount [extra disks](https://www.talos.dev/v1.7/reference/configuration/v1alpha1/config/#Config.machine.disks.), however, that wouldn't have an effect on the existing directory structure.

Is something like this possible? I don't want to run Talos as a whole on the external HDD, because I want to use that HDD for storage only, i.e. storing container images and a dedicated partition for Ceph (which will be used for persistent container volumes). Putting also the OS on there seems stretching the data pipelines on a Raspberry Pi. The SD cards are fast enough to run the container host OS. I don't want to have container images on the SD card because of the extra wear and tear this may cause.

**Comment by smira** [comment]
There is no great solution for that at the moment on Talos side.

The #8367 issue will eventually get there, but it might take some time to get it done.

**Reply by gnouts** [reply]
It's been two years, lot happened in the issue trail you mentioned, but I'm a bit confused if this is implemented or not now. The documentation only seems to direct to ImageCache functionality, which isn't exactly the initial ask here (if I understood correctly?).

So, Could you clarify the state of Talos regarding splitting /var volume ? Thanks

**Reply by smira** [reply]
It has been available for a long time now, https://docs.siderolabs.com/talos/v1.13/configure-your-talos-cluster/storage-and-disk-management/disk-management/system

Even more coming up in Talos 1.14, but you can totally have all the state on another disk while booting from SD.
